### PR TITLE
Enable metrics scraping of availability apps

### PR DESCRIPTION
**Pull request by ironcladlou**
Upgrade cluster-monitoring-operator and enable dynamic scraping of the
availability apps.

Blocked by https://github.com/openshift/openshift-ansible/pull/8531

### CODE REVIEW

**Comment by ironcladlou** [comment]
/cc @lukas-vlcek 

@elad661, your [mailing list thread](http://post-office.corp.redhat.com/archives/aos-online-monitoring/2018-May/msg00016.html) on bootstrapping will be relevant here now.

File: roles/openshift_monitor_availability/files/monitor-app-create.yaml [hunk 1]
```diff
@@ -101,6 +101,30 @@ objects:
         availabilityRoute: django-psql-persistent
         parameters: # Empty, use template defaults
 
+- apiVersion: monitoring.coreos.com/v1
+  kind: ServiceMonitor
+  metadata:
+    name: monitor-app-create
+    namespace: ${NAMESPACE}
+    labels:
+      k8s-app: monitor-app-create
```
[the remark this answers is not in the document]
Comment (ironcladlou) [reply roles/openshift_monitor_availability/files/monitor-app-create.yaml]: Agree, that would be a nice cleanup. Easy but numerous changes and more testing... want to just add that to the list of things for 3.11?

[the remark this answers is not in the document]
Comment (brancz) [reply roles/openshift_monitor_availability/files/monitor-app-create.yaml]: yeah, not urgent, especially as the "all" selector would effectively be backward compatible

**Comment by ironcladlou** [comment]
I think https://github.com/openshift/openshift-ansible/pull/8542 needs to go in with this one to avoid flakiness.

**Pull request merged**